pkg/scheduler/internal/cache/node_tree.go

```diff
@@ -62,8 +62,13 @@ func newNodeTree(nodes []*v1.Node) *NodeTree {
 	nt := &NodeTree{
 		tree: make(map[string]*nodeArray),
 	}
+	if len(nodes) == 0 {
+		return nt
+	}
+	nt.mu.Lock()
+	defer nt.mu.Unlock()
```

Comment:
You don't need to acquire the lock at all, at this point nt is still a local variable.